### PR TITLE
feature: support root ("/") as path

### PR DESCRIPTION
**Pull request by deluxghost**
currently we cannot set "/" as the path option, because after string concat, many urls will be like `href="//oauth2"`, which somehow equals to invalid link `http://oauth` (should be `/oauth2`)

`/auth` + `/oauth2` => `/auth/oauth2`
`/` + '/oauth2' => `//oauth2`

`path.Join` can handle `/`s properly

Changes:

* replace string concat to `path.Join`
* add `pathjoin` func to template funcMap

### CODE REVIEW

**Comment by greenpau** [comment]
@deluxghost , thank you for the contribution! I will review next week. Also, I will also add CONTRIBUTING.md.

File: pkg/handlers/settings.go [hunk 1]
```diff
@@ -26,7 +27,8 @@ func ServeSettings(w http.ResponseWriter, r *http.Request, opts map[string]inter
 	claims := opts["user_claims"].(*jwt.UserClaims)
 	uiFactory := opts["ui"].(*ui.UserInterfaceFactory)
 	view := strings.TrimPrefix(r.URL.Path, authURLPath)
-	view = strings.TrimPrefix(view, "/settings")
+	view = strings.TrimPrefix(view, "/")
```
[the remark this answers is not in the document]
Comment (deluxghost) [reply pkg/handlers/settings.go:30]: here i wrote an example: https://play.golang.org/p/Y5nz4rbCS-b

[the remark this answers is not in the document]
Comment (greenpau) [reply pkg/handlers/settings.go:30]: @deluxghost, LGTM! Please read contributing guide, add CLA consent and signoff to your commit message.

**Comment by deluxghost** [comment]
am i doing right?

**Comment by greenpau** [comment]
@deluxghost , cla is fine, but you need to sign off, i.e. “git -s ...”

See https://medium.com/@MarkEmeis/git-commit-signoff-vs-signing-9f37ee272b14

**Comment by greenpau** [comment]
@deluxghost , thank you for the contribution! 👍

**Comment by greenpau** [comment]
@deluxghost , the following change broke QR Code image. I am going to change it back for now. 

https://github.com/greenpau/caddy-auth-portal/pull/45/files#diff-ebd9c76d4dc88e595d703b4e915581c96d0d93e46c27a66a24db4999499038a1R168

**Comment by deluxghost** [comment]
> @deluxghost , the following change broke QR Code image. I am going to change it back for now.
> 
> https://github.com/greenpau/caddy-auth-portal/pull/45/files#diff-ebd9c76d4dc88e595d703b4e915581c96d0d93e46c27a66a24db4999499038a1R168

This might work (untested):

`{{ pathjoin .ActionEndpoint "/settings/mfa/barcode/" .Data.code_uri_encoded }}.png`

**Pull request merged**